library/std/src/ffi/c_str.rs

```diff
@@ -328,6 +328,17 @@ impl FromVecWithNulError {
     }
 }
 
+/// An error indicating that no nul byte was present.
+///
+/// A slice used to create a [`CStr`] must contain a nul byte somewhere
+/// within the slice.
+///
+/// This error is created by the [`CStr::from_bytes`] method.
+///
+#[derive(Clone, PartialEq, Eq, Debug)]
+#[unstable(feature = "tolerant_cstr_from_bytes", issue = "none")]
+pub struct FromBytesError;
```

Comment:
If you define this without any private members, anyone can create this struct, which we usually prevent to make sure we can change things in the future. Check some of the other error types in std, and you'll see some have a single private `()` member because of this.

And unrelated to that: error types should implement `Error` (and therefore also `Display`).

Reply:
Another way of doing this is to use `#[non_exhaustive]`